Update documentation to include experimental features (#2970)
modin-project/modin: docs/experimental_features/index.rst

Overview
========
Modin aims to not only optimize Pandas, but also provide a comprehensive, 
integrated toolkit for data scientists. We are actively developing data science tools 
such as DataFrame - spreadsheet integration, DataFrame algebra, progress bars, SQL queries

> on DataFrames, and more. Please feel free to contact us anytime for the latest updates!

Reviewer: ```suggestion
on DataFrames, and more. Join the Discourse_ for the latest updates!
```

> Modin Spreadsheets API: Render Dataframes as Spreadsheets

Reviewer: ```suggestion
Modin Spreadsheet API: Render Dataframes as Spreadsheets
```
